### ISSUE TITLE
source push (Old & Legacy) is not working as intended

### ISSUE THREAD

**Issue by Blaada** [forcedotcom/cli#1437]
I did the switch in march 3 to the new beta version, today I upgraded the cli and tested old source push `sfdx force:source:push` and `sfdx force:source:legacy:push` but everytime i get this issue `ERROR running force:source:legacy:push:  The "path" argument must be of type string. Received null`

The `sfdx force:source:beta:push` is working but still not displaying Error message + it's not deploying some metadata.

FYI : i'm using `-f` option in all of these commands.

can someone please help on this ?

**Comment by maggiben** [comment]
Hi @Blaada Q: Is the new command: `sfdx force:source:beta:push` not deploying some metadata from your org ? can you please post a way to reproduce this error ? Same thing for the legacy and old commands.

**Comment by jmccotter** [comment]
I am getting this same error with a large deploy using `sfdx force:source:beta:push -f`.  Mine was working with the last version of SFDX and legacy push but fails to work with beta push.  I have not tested legacy push with SFDX 7.143 yet.  Here are the errors with all of my info redacted in beta push:
```
DEPLOY PROGRESS | ████████████████████████████████████████ | 6557/6557 Components
  @oclif/config start postdeploy hook +23m
  @oclif/config postdeploy hook done +1ms
  sfdx:core TRACE Setup child 'ComponentSet' logger instance +18m
  sfdx:SourceTracking DEBUG ensureRemoteTracking: remote tracking does not exist yet; getting instance +23m
  sfdx:core TRACE Setup child 'RemoteSourceTrackingService' logger instance +807ms
  sfdx:core TRACE Setup child 'ComponentSet' logger instance +226ms
  sfdx:core TRACE Setup child 'Org' logger instance +162ms
  sfdx:core TRACE Setup child 'Aliases' logger instance +0ms
  sfdx:Aliases INFO Reading config file: /Users/************/.sfdx/alias.json +26m
ERROR running force:source:beta:push:  The "path" argument must be of type string. Received undefined 
*** Internal Diagnostic ***

TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received undefined
    at validateString (internal/validators.js:120:11)
    at dirname (path.js:1128:5)
    at VirtualTreeContainer.exists (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@salesforce/source-deploy-retrieve/lib/src/resolve/treeContainers.js:186:56)
    at MixedContentSourceAdapter.populate (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@salesforce/source-deploy-retrieve/lib/src/resolve/adapters/mixedContentSourceAdapter.js:52:24)
    at MixedContentSourceAdapter.getComponent (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@salesforce/source-deploy-retrieve/lib/src/resolve/adapters/baseSourceAdapter.js:50:21)
    at MetadataResolver.resolveComponent (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@salesforce/source-deploy-retrieve/lib/src/resolve/metadataResolver.js:122:44)
    at MetadataResolver.getComponentsFromPathRecursive (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@salesforce/source-deploy-retrieve/lib/src/resolve/metadataResolver.js:79:40)
    at MetadataResolver.getComponentsFromPathRecursive (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@salesforce/source-deploy-retrieve/lib/src/resolve/metadataResolver.js:102:37)
    at MetadataResolver.getComponentsFromPathRecursive (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@salesforce/source-deploy-retrieve/lib/src/resolve/metadataResolver.js:102:37)
    at MetadataResolver.getComponentsFromPathRecursive (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@salesforce/source-deploy-retrieve/lib/src/resolve/metadataResolver.js:102:37)
Outer stack:
    at Function.wrap (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@salesforce/core/lib/sfdxError.js:171:27)
    at Push.catch (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@salesforce/command/lib/sfdxCommand.js:248:67)
    at runMicrotasks (<anonymous>)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async Push._run (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@salesforce/command/lib/sfdxCommand.js:85:13)
    at async Config.runCommand (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@oclif/config/lib/config.js:173:24)
    at async SfdxMain.run (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@oclif/command/lib/main.js:27:9)
    at async SfdxMain._run (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/node_modules/@oclif/command/lib/command.js:43:20)
    at async Object.run (/Users/************/.nvm/versions/node/v12.22.0/lib/node_modules/sfdx-cli/dist/cli.js:162:47)
******
  ```
Then further down during the updating source tracking step, it is hitting another error with a couple fieldsets on the same object:
```
  sfdx:connection DEBUG >>> Query start >>> +1s
  sfdx:connection DEBUG SOQL = SELECT MemberType, MemberName, IsNameObsolete, RevisionCounter FROM SourceMember WHERE RevisionCounter > 46175 +0ms
  sfdx:connection DEBUG request: {"method":"GET","url":"https://*********.my.salesforce.com/services/data/v54.0/tooling/query?q=SELECT%20MemberType%2C%20MemberName%2C%20IsNameObsolete%2C%20RevisionCounter%20FROM%20SourceMember%20WHERE%20RevisionCounter%20%3E%2046175","headers":{"content-type":"application/json","user-agent":"sfdx toolbelt:"}} +0ms  
sfdx:RemoteSourceTrackingService WARN Polling for SourceMembers timed out after 50 attempts (last 48 were empty) ) +1s
  sfdx:RemoteSourceTrackingService DEBUG Could not find 2 SourceMembers: FieldSet__************__c.************,[object Object],FieldSet__*********__c.*********,[object Object] +0ms
```

Here's my sfdx verbose version if that's helpful:
```
CLI Version : 
        sfdx-cli/7.143.0

 Architecture: 
        darwin-x64

 Node Version : 
        node-v12.22.0

 Plugin Version: 
        @oclif/plugin-autocomplete 0.3.0 (core)
        @oclif/plugin-commands 1.3.0 (core)
        @oclif/plugin-help 3.3.1 (core)
        @oclif/plugin-not-found 1.2.6 (core)
        @oclif/plugin-plugins 1.10.11 (core)
        @oclif/plugin-update 1.5.0 (core)
        @oclif/plugin-warn-if-update-available 1.7.3 (core)
        @oclif/plugin-which 1.0.4 (core)
        @salesforce/sfdx-plugin-lwc-test 0.1.7 (core)
        alias 1.2.1 (core)
        apex 0.11.0 (core)
        auth 1.8.1 (core)
        community 1.1.4 (core)
        config 1.3.23 (core)
        custom-metadata 1.0.12 (core)
        data 0.6.10 (core)
        generator 1.2.2 (core)
        info 1.3.1 (core)
        limits 1.3.0 (core)
        org 1.11.2 (core)
        salesforce-alm 54.0.2 (core)
        schema 1.1.0 (core)
        sfdx-cli 7.143.0 (core)
        source 1.8.19 (core)
        telemetry 1.4.0 (core)
        templates 54.2.0 (core)
        trust 1.1.0 (core)
        user 1.7.1 (core)

 OS and Version: 
        Darwin 20.6.0
```
The force:source:beta:status command also errors out, but I figured that should be opened as a separate issue from the push issue.

**Comment by jmccotter** [comment]
@Blaada if you're still having this issue with beta push, could you respond with your issue details so they remove the more info needed and don't auto-close it.
